notebooks/rocoto.ipynb

```diff
@@ -0,0 +1,1013 @@
+      "  <cycledef>202410290000 202410300000 06:00:00</cycledef>\n",
+      "  <log>logs/test.log</log>\n",
+      "  <task name=\"greet\">\n",
+      "    <cores>1</cores>\n",
+      "    <walltime>00:00:10</walltime>\n",
+      "    <command>echo Hello, World!</command>\n",
+      "    <jobname>greet</jobname>\n",
+      "  </task>\n",
+      "</workflow>\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/simple-workflow.xml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39f50dbd-e3f3-4e1e-a10b-f622f635b16b",
+   "metadata": {},
+   "source": [
+    "The following is a workflow that is missing required components. `workflow` doesn't contain a `realtime` attribute, a `log:` block isn't included, and `task_greet` doesn't include a `command`.\n",
+    "<!--cell 10-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "56d04e6b-5fa2-4c93-ac2f-c95fa23c888e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "workflow:\n",
+      "  attrs:\n",
+      "    scheduler: slurm\n",
+      "  cycledef:\n",
+      "    - spec: 202410290000 202410300000 06:00:00\n",
+      "  tasks:\n",
+      "    task_greet:\n",
+      "      cores: 1\n",
+      "      walltime: 00:00:10\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/rocoto/err-workflow.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f102c6ca-c8a5-4f9c-8e40-b5fa040bbeab",
+   "metadata": {},
+   "source": [
+    "When any type of validation error occurs, `realize()` will raise an exception indicating what type of error occurred. The number of validation errors present and their locations within the workflow structure are given.\n",
+    "<!--cell 12-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "16704f19-cbca-4765-8c72-16512fc96e9b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-10-28T12:34:55]    ERROR 3 UW schema-validation errors found\n",
+      "[2024-10-28T12:34:55]    ERROR Error at workflow -> attrs:\n",
+      "[2024-10-28T12:34:55]    ERROR   'realtime' is a required property\n",
+      "[2024-10-28T12:34:55]    ERROR Error at workflow -> tasks -> task_greet:\n",
+      "[2024-10-28T12:34:55]    ERROR   'command' is a required property\n",
+      "[2024-10-28T12:34:55]    ERROR Error at workflow:\n",
+      "[2024-10-28T12:34:55]    ERROR   'log' is a required property\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "YAML validation errors\n"
+     ]
+    }
+   ],
+   "source": [
+    "try:\n",
+    "    rocoto.realize(\n",
+    "        config=Path('fixtures/rocoto/err-workflow.yaml'),\n",
+    "        output_file='tmp/err-workflow.xml'\n",
+    "    )\n",
+    "except Exception as e:\n",
+    "    print(e)"
```

Comment:
```suggestion
    "    print(e, type(e))"
```
Users would generally want to catch the `UWConfigError` exception that is raised, so that a different type of exception (say, `DivideByZeroError`) would bubble up. To catch `UWConfigError` we'd also have to `import` it, so I thought maybe it would be instructive enough here to just show that UW exceptions have their own types to differentiate them from built-in Python or other exceptions.

Reply:
Good idea, I added a sentence to the markdown too to bring attention to the error type as well.